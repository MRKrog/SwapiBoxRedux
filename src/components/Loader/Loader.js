import React from 'react';
import PropTypes from 'prop-types';

import gear from '../../images/gear-logo.png';

const Loader = ({ loading }) => {
  const loadingClass = loading ? 'loading-true' : null;
  return (
    <div className='logo-images'>
      <img src={gear} className={loadingClass} alt="logo" />
    </div>
  )
}

Loader.propTypes = {
  loading: PropTypes.bool
}

export default Loader;
