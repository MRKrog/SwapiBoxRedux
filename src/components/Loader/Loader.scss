
.logo-images {
  display: block;
  position: relative;
  margin: auto;
  img {
    height: 300px;
  }
}

.loading-true {
  animation: App-logo-spin infinite 20s linear;
  z-index: 2;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
