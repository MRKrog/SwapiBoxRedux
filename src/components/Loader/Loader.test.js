import React from 'react'
import Loader from './Loader'
import { shallow } from 'enzyme';

describe('Loader', () => {
  let wrapper;
  let loadStatus = true;

  beforeEach(() => {
    wrapper = shallow(<Loader loading={loadStatus} />)
  })

  it('should match the snapshot when rendered', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have a class of loading-true if loading prop is equal to true', () => {
    const favoriteImage = wrapper.find('img')
    // console.log(favoriteImage);
    expect(favoriteImage.hasClass('loading-true')).toBe(true);
  })

  it('should not have a class loading-true is loading prop is equal to false', () => {
    wrapper.setProps({ loading: false })
    const favoriteImage = wrapper.find('img')
    expect(favoriteImage.hasClass('loading-true')).toBe(false)
  })

})
