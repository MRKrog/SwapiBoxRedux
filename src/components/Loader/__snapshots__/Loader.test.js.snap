// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should match the snapshot when rendered 1`] = `
<div
  className="logo-images"
>
  <img
    alt="logo"
    className="loading-true"
    src="gear-logo.png"
  />
</div>
`;
